src/charm_state.py

```diff
@@ -266,6 +268,35 @@ def userids_to_list(cls, value: str) -> typing.List[str]:
                 raise ValidationError(f"Invalid user ID format: {user_id}", cls)
         return value_list
 
+    @validator("ready_check")
+    @classmethod
+    def to_pebble_check(cls, value: str) -> typing.Dict[str, typing.Union[str, int]]:
+        """Convert the ready_check field to pebble check.
+
+        Args:
+            value: the input value.
+
+        Returns:
+            The pebble check.
+
+        Raises:
+            ValidationError: if ready_check is invalid.
+        """
+        # expected
+        # period;threshold;timeout
+        config_values = value.split(",")
+        if len(config_values) < 3:
+            raise ValidationError(f"Invalid ready_check, less than 3 values: {value}", cls)
+        try:
+            period = config_values[0]
+            threshold = int(config_values[1])
+            timeout = config_values[2]
+            return {"period": period, "threshold": threshold, "timeout": timeout}
```

Comment:
Just out of curiosity, i'm wondering if it would help to check the time units here as well (`values[-1] in ("s", "m", "h")`) :)
One minor thing I also like to do is to `str.trim()` them just incase the user does `config1, config2, config3` (spaces inbetween for aesthetics)

Reply:
Curious, I thought that I added...I'll add it now, thanks!